src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,259 @@
+            } catch(IndexOutOfBoundsException unMarkOutOfBounds) {
+                System.out.println("Unmark index is out of bounds!!");
+            }
+            break;
+        case COMMAND_DELETE:
+            int deleteIndex = Integer.parseInt(commandArr[1]) - 1;
+            try {
+                itemNum--;
+                System.out.println("Noted. I've removed this task:\n" + taskArrayList.get(deleteIndex) +
+                        "\nNow you have " + itemNum + " tasks in the list.");
+                taskArrayList.remove(deleteIndex);
+            } catch(IndexOutOfBoundsException deleteIndexOutOFBounds) {
+                System.out.println("The item you want to delete is not in the list!");
+            }
+            break;
+        case COMMAND_BYE:
+            exitProgram();
+            break;
+        default:
+            addTask(inputCommand);
+            break;
+        }
+    }
+
+    private static void saveTaskList() throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        for (Task a : taskArrayList) {
+            fw.write(String.valueOf(a.getTaskType()) + "-" + String.valueOf(a.isDone()) + "-"
+                    + a.saveDescription() + a.saveAdditionalInfo() + System.lineSeparator());
+        }
+        fw.close();
+    }
+
+    private static void exitProgram() {
+        System.out.println("Bye. Hope to see you again soon!");
+        System.exit(0);
+    }
+
+    private static void addTask(String inCommand) throws DukeException {
+        String[] sentenceArr = inCommand.split(" ", 2);
+        switch (sentenceArr[0]) {
+        case COMMAND_TODO:
+            try {
+                taskArrayList.add(new ToDo(sentenceArr[1]));
+                printAcknowledgeAddMessage();
+                itemNum++;
+            } catch (IndexOutOfBoundsException todoEmpty) {
+                System.out.println("OOPS!!! The description of todo cannot be empty.");
+            }
+            break;
+        case COMMAND_DEADLINE:
+            try {
+                dateArray = sentenceArr[1].split("/by", 2);
+                if (dateArray[1].length() == 0 || dateArray[0].length() == 0) {
+                    throw new DukeException();
+                }
+                taskArrayList.add(new Deadline(dateArray[0], dateArray[1]));
+                printAcknowledgeAddMessage();
+                itemNum++;
+            } catch (ArrayIndexOutOfBoundsException deadlineEmpty) {
+                System.out.println("OOPS!!! The description and date of deadline cannot be empty.");
+            } catch (DukeException invalidDateInput) {
+                System.out.println("Invalid deadline input! Please try again.");
+            }
+            break;
+        case COMMAND_EVENT:
+            try {
+                eventArray = sentenceArr[1].split("/at", 2);
+                if (eventArray[1].length() == 0 || eventArray[0].length() == 0) {
```

Comment:
This is a [compound boolean expression](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions). How refactoring it into in intermediate steps with named constants?